woocommerce/woocommerce
# Support installing live branches from the manifest

### All Submissions:






### Changes proposed in this Pull Request:

This adds support to install and activate WooCommerce plugin versions acquired from the live branches manifest.

I have intentionally kept the UI quite simple still to avoid a lot of complexity and focus on the main features of allowing the install and activation of any plugin build from the live branches manifest.






### How to test the changes in this Pull Request:



0. Add the `register_page` code back here: https://github.com/woocommerce/woocommerce/blob/d8b0a26b6e4d151030457f0761e918a6bd1fcdac/plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-live-branches.php#L23 (since we don't have feature flags)
1. Run this branch alongside an activated WooCommerce core plugin
2. Go to the menu WooCommerce > Live Branches
3. A list of live branches you can install should be listed. If they are not already installed they should display a button "Install and Activate"
4. Click "Install and Activate" on a plugin of your choice. A spinner should appear while it is downloading and activating the plugin.
5. Eventually this install should finish and the item you clicked should now show a greyed out button "Activated"
6. To swap to a different live branch you can click "Install and Activate" on another list item.
7. This should successfully install and swap to the live branch you just clicked.
8. Now if you do a page refresh (I think keeping track of the install state of all plugins could be a future enhancement? I wanted to keep scope small as possible) you should see one of your live branches is installed but not activated because it has a button that just says "Activate"
9. Click "Activate" and it should just perform activation for that plugin and deactivate the old one.


<img width="1125" alt="Screenshot 2022-12-19 at 6 56 35 PM" src="https://user-images.githubusercontent.com/1281828/208357628-fcc62a0e-b620-42cf-b847-f34ec977b466.png">


### Other information:




### FOR PR REVIEWER ONLY:
